### ISSUE TITLE
Can't find subsequent children ref of a refCollection

### ISSUE THREAD

**Issue by goncalvesjoao** [QubitProducts/react-test-tree#46]
Guys, I've got the following test that I can't make it pass, for the life of me.
Am I doing something wrong?

```javascript
const StarwarsDroids = React.createClass({

  getInitialState() {
    return {
      droids: [
        { id: 1, name: 'r2d2', avatar: 'none' },
        { id: 2, name: 'c-3po', avatar: 'none' },
      ],
    };
  },

  render() {
    return (
      <ul className='list-inline' refCollection='droids'>
        { this.state.droids.map(this.renderDroid) }
      </ul>
    );
  },

  renderDroid(droid, index) {
    return (
      <li key={ index }>
        <img ref='avatar' className='img-circle' src={ droid.avatar } />
      </li>
    );
  },

});

describe('StarwarsDroids', () => {
  context('when no props are used', () => {
    it('should list all droids with an avatar', () => {
      const starwarsDroidsTree = TestTree(<StarwarsDroids />);

      expect(starwarsDroidsTree.droids.length).to.equal(2);

      starwarsDroidsTree.droids.forEach((droid) => {
        expect(droid.avatar.getAttribute('src')).to.equal('none');
      });
    });
  });
});
```

droid.avatar is undefined, and droid.refs is empty. What am I missing?

Thanks in advance!!

**Comment by oliverwoodings** [comment]
I think you are mis-understanding how refs are 'owned' in a component. Refs belong to the component that outputs them from it's `render` method. If we simplify your render function, this is what you are actually returning:

```
const StarwarsDroids = React.createClass({

  render() {
    return (
      <ul className='list-inline' refCollection='droids'>
        <li>
          <img ref='avatar' className='img-circle' src='none' />
        </li>
        <li>
          <img ref='avatar' className='img-circle' src='none' />
        </li>
      </ul>
    );
  }

});
```

The problem should be fairly clear now; your `avatar` ref actually belongs to the `StarwarsDroid` component, and you are declaring it multiple times which means it will just get overwritten. If you check `starwarsDroidsTree.avatar` in your test you should see it points to your second `img` element.

If you want to use refs in that way, I suggest you abstract out your individual droids into a separate component called `StarwarsDroid`. This is pretty good practice anyway :)

tl;dr this is not the bug you are looking for

![](http://media0.giphy.com/media/11jaBqRgruK3Xq/200w.gif)

**Comment by goncalvesjoao** [comment]
lol thanks for the reply @oliverwoodings makes sense!
